### ISSUE TITLE
Kill running ffmpeg processes when sources change

### ISSUE THREAD

**Issue by ChairmanMeow** [Jalle19/node-ffmpeg-mpegts-proxy#47]
For my channels, I swap out the source via . python script:

    {
        "url": "/channel2",
        "source": "rtmp://URLaWhichIchangetoURLb",
        "name": "2",
        "provider": "Provider Two"
    }

however, for some reason I am seeing discrepancies.. a few zombie processes are created, and it seems like the previous avcov PID from the previous source. Do you know what is causing this issue?

There are several avcov processes that are pulling from the same source which is confusing since I only have 1 

![image](https://user-images.githubusercontent.com/1448407/44297105-91c63600-a280-11e8-9225-945e1126731b.png)

**Comment by Jalle19** [comment]
Do you mean those processes are started right when you change your source? Or are those existing streams that were open when you changed the source, and you would like them to disconnect automatically when the source changes?

**Comment by ChairmanMeow** [comment]
wow! really appreciate the fast reply. 
yes I plan to use this with 20+ sources lol.

To clarify, In the screenshot above, I am only testing with 4 sources .  But i see 7 avconv PID which is confusing. 
When i tried with 15 sources, not sure what happened, but noticed a lot of zombie processes were spawned.

I am still looking into this, and temporarily i have to stop the service, and restart it to get it back to working order.

**Comment by Jalle19** [comment]
Does it spawn these processes even if you don't play anything?

**Comment by ChairmanMeow** [comment]
Quick update - I found the issue...
I am constantly pulling the URL via my CDN.
However, when I change my sourceURL, the avconv process is not killed. My CDN is still pulling from the old avconv PID.

I think it is best to add that if the source changes, then kill the existing avconv thread, which should cause anyone pulling that URL to disconnect and then reconnect to the new source.

Does that sound right?

**Comment by Jalle19** [comment]
Yeah that sounds reasonable. Do you want to take a stab at this yourself or?

**Comment by ChairmanMeow** [comment]
do you mind making the change? my node skills aren't that good. From looking at the code, i see:
```
var onSourceChange = function() {
	winston.info('Source definitions have changed, reloading ...');
};
```
Wondering if i should add a `stream.kill()` in there? Is it possible to detect which channels had their source changed and to kill the old respective avconv processes?

**Comment by Jalle19** [comment]
I guess I can take a look

**Comment by ChairmanMeow** [comment]
Hey any updates on this? Can you get to it soon? Can send ETH/BTC over 👍

**Comment by Jalle19** [comment]
Sorry, kinda forgot abou this. Let's see.

**Comment by ChairmanMeow** [comment]
Any updates on this?

**Comment by Jalle19** [comment]
Haven't had time to take a look, feel free to try yourself!
